scipy/optimize/_qap.py

```diff
@@ -0,0 +1,389 @@
+
+
+def quadratic_assignment(
+    cost_matrix,
+    dist_matrix,
+    seed_cost=[],
+    seed_dist=[],
+    maximize=False,
+    n_init=1,
+    init_method="barycenter",
+    max_iter=30,
+    shuffle_input=True,
+    eps=0.05,
+):
+    """
+    Solve the quadratic assignment problem
+
+    This class solves the Quadratic Assignment Problem and the Graph
+    Matching Problem (QAP) through an implementation of the Fast
+    Approximate QAP Algorithm (FAQ) (these two problems are the same up
+    to a sign change) [1].
+
+    This algorithm can be thought of as finding an alignment of the
+    vertices of two graphs which minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2].
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square, positive
+        A square adjacency matrix
+
+    dist_matrix : 2d-array, square, positive
+        A square adjacency matrix
+
+    seed_cost : 1d-array, optional, (default = [])
+        An array where each entry is an index of a node in 'cost_matrix'
+        (shape (m , 1) where m <= number of nodes).
+
+    seeds_dist : 1d-array, optional, (default = [])
+        An array where each entry is an index of a node in 'dist_matrix'
+        The elements of 'seed_cost' and 'seed_dist' are vertices which
+        are known to be matched, that is, 'seed_cost[i]' is matched to
+        vertex 'seed_dist[i]'. Array shape (m , 1)
+        where m <= number of nodes
+
+    maximize : bool (default = False)
+        Gives users the option to solve the Graph Matching Problem (GMP)
+        rather than QAP. This is accomplished through trivial negation
+        of the objective function.
+
+
+    options : dict, optional
+        A dictionary of solver options. All methods accept the following
+        options:
+
+            n_init : int, positive (default = 1)
+               Number of random initializations of the starting
+               permutation matrix that the FAQ algorithm will undergo.
+               n_init automatically set to 1 if init_method = 'barycenter'
+            init_method : string (default = 'barycenter')
+               The initial position chosen:
+               "barycenter" : the non-informative "flat doubly stochastic
+               matrix,":math:'J=1*1^T /n' , i.e the barycenter of the
+               feasible region
+               "rand" : some random point near :math:'J, (J+K)/2', where K
+               is some random doubly stochastic matrix
+            max_iter : int, positive (default = 30)
+               Integer specifying the max number of Franke-Wolfe iterations.
+               FAQ typically converges with modest number of iterations.
+            shuffle_input : bool (default = True)
+               Gives users the option to shuffle the nodes of 'cost_matrix'
```

Comment:
I don't think I understand what this does. "avoid results from inputs that were already matched" suggests that we don't want to include those nodes in the calculation of the cost, but  I can't tell how "shuffling" (which sounds like random shuffling) would prevent this.

Reply:
Hi @mdhaber, I've worked with Ali on some of this work - in the past we have had instances where matching results could be artificially low/high due to the starting permutation of the cost matrices/graphs. The goal here was to have an option to "pre-shuffle" the input to avoid this problem (and unshuffle at the end to make sure the output matches what the user input).

Reply:
OK, it sounds like I misunderstood what this did, so I imagine other users might. Can you try rewording this so that the user can better judge whether to leave this on or turn it off?